components/dashboard/src/user-settings/Preferences.tsx

```diff
@@ -142,6 +170,37 @@ export default function Preferences() {
                         </p>
                     </div>
                 </div>
+
+                <h3 className="mt-12">Timeout </h3>
+                <p className="text-base text-gray-500 dark:text-gray-400">Customize timeout setting for workspace.</p>
+                <div className="mt-4 max-w-xl">
+                    <h4>Workspace timeout</h4>
+                    <span className="flex">
+                        <input
+                            type="text"
+                            className="w-96 h-9"
+                            value={workspaceTimeout}
+                            placeholder="timeout time, such as 30m, 1h, max 24h"
+                            onChange={(e) => setWorkspaceTimeout(e.target.value)}
+                        />
+                        <button
+                            className="secondary ml-2"
+                            onClick={() => actuallySetWorkspaceTimeout(workspaceTimeout)}
+                        >
+                            Save Changes
+                        </button>
+                    </span>
+                    <div className="mt-1">
+                        <p className="text-gray-500 dark:text-gray-400">some description</p>
+                    </div>
+
+                    <CheckBox
+                        title="Disabled Close Timeout"
+                        desc={<span></span>}
+                        checked={disabledClosedTimeout}
+                        onChange={(e) => actuallySetDisabledClosedTimeout(e.target.checked)}
+                    />
```

Comment:
**question**: What does this do? It seems to not work. Do we need it?

Reply:
Ah, it seems to work now. I could not toggle the checkbox when I tried earlier. 🤷 

Still, what does this checkbox do? Do we need it?

Reply:
@iQQBot Thanks for your patience while I'm trying to understand!

❓ Does this mean you will no longer be able to configure a different default inactivity timeout? I'm definitely missing something, but could you explain a bit more the use cases where this checkbox would be useful?

Reply:
Ah, ok! Now it makes sense! Thanks for taking the time to clarify this, @iQQBot! 🌮 🌮 

I also noticed that we now have a **[docs section](https://www.gitpod.io/docs/configure/workspaces/workspace-lifecycle#extending-timeouts:~:text=Currently%2C%20all%20timeouts%20are%20dependent%20on%20an%20active%20editor%20connection.%20Closing%20your%20Gitpod%20connected%20editor%20or%20IDE%20will%20reduce%20the%20workspace%20timeout%20to%205%20minutes.)** about this, where we call it editor disconnect, thanks to https://github.com/gitpod-io/website/pull/3326 (Cc @loujaybee @Siddhant-K-code @laushinka @akosyakov)!

> **Editor Disconnect**
> Currently, all timeouts are dependent on an active editor connection. Closing your Gitpod connected editor or IDE will reduce the workspace timeout to 5 minutes.

---

Re-thinking how this works and how users understand this feature, what do you think of rephrasing the copy like the following for simplicity and brevity?

| Timeouts |
|-|
| ![Inactivity Timeout](https://user-images.githubusercontent.com/120486/221001911-7f1a770b-928c-4e01-87b4-214ce6b7d3dd.png) |

Reply:
Thanks @iQQBot! Just highlighting some copy changes from the suggestion above to simplify and improve readability. What do you think? 

| BEFORE | AFTER |
|-|-|
| <img width="776" alt="timeouts-before" src="https://user-images.githubusercontent.com/120486/221242456-4f59d5e7-84e1-443e-8bbf-2b91085c9200.png"> | <img width="776" alt="timeouts-after" src="https://user-images.githubusercontent.com/120486/221242467-2c6b07fc-4fd5-4ceb-8886-819168c42237.png"> |

💡 **tip**: Also, using `font-semibold` instead of `bold` can improve the rendering here. 

❗ issue: This is also missing some validation states where a user enters a blank value or an invalid value but relying on the browser alert sounds ok for this iteration. Could you open a follow-up issue for doing proper form validation in this case?

Reply:
> This 5m close timeout only for gitpod.io. Other installation may have different value, and it set in workspace cluster, so server actually don't know it

In Dedicated, this would be exposed through the organisation settings, can't we assume 5 minutes for now as default until we have also this configuration at the admin/organisation level? 

- https://github.com/gitpod-io/gitpod/issues/16090

Reply:
The workspace inactivity timeout copy + behaviour looks good, the editor timeout needs some refinement. I have [created an internal RFC](https://www.notion.so/gitpod/Naming-of-Editor-Timeout-Feature-f209ede9b9754bb0853ea40ecc894e4f?pvs=4) to resolve some questions about the naming of the editor timeout copy text and naming. My intention is to **get feedback and close that decision in the next 24 hours** to unblock this PR. 

However, given this PR is solving two distinct product issues: 

- https://github.com/gitpod-io/gitpod/issues/16200
- https://github.com/gitpod-io/gitpod/issues/16089

If we are keen to merge it, we should consider either commenting out/disabling the editor timeout part of this PR + raising a separate PR for the editor timeout, whilst we resolve naming + copy decisions. Or wait until that decision and merge both at once.